### PR TITLE
Rename Orgrepo and Path

### PR DESCRIPTION
**Pull request by kishorerj**
Changes Refactor repospec Issue :https://github.com/kubernetes-sigs/kustomize/issues/4866.             
rename OrgRepo to RepoPath. Rename Path to  KustRootPath to clarify that it isn't the repo or clone root

### CODE REVIEW

File: api/internal/git/repospec.go [hunk 1]
```diff
@@ -30,16 +30,16 @@ type RepoSpec struct {
 	// Host, e.g. https://github.com/
 	Host string
 
-	// orgRepo name (organization/repoName),
+	// RepoPath name (organization/repoName),
```
Comment (annasong20) [inline api/internal/git/repospec.go:33]: nit: Can we change the doc string for `RepoPath` to be more descriptive? For instance, 

`RepoPath` is the relative path to the repository on `Host`. 

Right now it still sounds too specific to `OrgRepo`.

File: api/internal/git/repospec.go [hunk 2]
```diff
@@ -30,16 +30,16 @@ type RepoSpec struct {
...
 	// e.g. kubernetes-sigs/kustomize
-	OrgRepo string
+	RepoPath string
 
-	// Dir where the orgRepo is cloned to.
+	// Dir where the RepoPath is cloned to.
```
Comment (annasong20) [inline api/internal/git/repospec.go:37]: super nit: Technically, `Dir` is `where the repository is cloned to`. The path to the repository isn't actually cloned anywhere.

File: api/internal/git/repospec_test.go [hunk 3]
```diff
@@ -54,16 +54,16 @@ func TestNewRepoSpecFromUrl_Permute(t *testing.T) {
...
 					t.Run(fmt.Sprintf("t%d", i), func(t *testing.T) {
-						uri := makeURL(hostRaw, orgRepo, pathName, hrefArg)
+						uri := makeURL(hostRaw, repoPath, pathName, hrefArg)
 						rs, err := NewRepoSpecFromURL(uri)
 						require.NoErrorf(t, err, "unexpected error creating RepoSpec for uri %s", uri)
 						assert.Equal(t, hostSpec, rs.Host, "unexpected host for uri %s", uri)
-						assert.Equal(t, orgRepo, rs.OrgRepo, "unexpected orgRepo for uri %s", uri)
-						assert.Equal(t, pathName, rs.Path, "unexpected path for uri %s", uri)
+						assert.Equal(t, repoPath, rs.RepoPath, "unexpected RepoPath for uri %s", uri)
+						assert.Equal(t, pathName, rs.KustRootPath, "unexpected path for uri %s", uri)
```
Comment (annasong20) [inline api/internal/git/repospec_test.go:66]: super nit: Might be helpful to clarify `KustRootPath` here.

**Review by annasong20 — COMMENTED** [review]
Thanks for this PR! Looks great! I left some optional comments.

**Comment by annasong20** [comment]
@kishorerj If you click on "Details" (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kustomize/4922/kustomize-presubmit-master/1610853341464629248) to the right of the failed test, you'll see that the problem is

```
internal/localizer/util.go:179:73: repoSpec.OrgRepo undefined (type *git.RepoSpec has no field or method OrgRepo)
```

I merged new code that uses `OrgRepo`. If you change these occurrences of `OrgRepo` to your updated name, this problem should be fixed.

**Comment by kishorerj** [comment]
@annasong20 @KnVerey @natasha41575 Fixed the test failure because of further commits..Please review

**Pull request merged**